packages/realm_dart/lib/src/native/realm_core.dart

```diff
@@ -3030,6 +3034,10 @@ class _RealmCore {
           collectionHandle = mapHandle;
 
           final map = realm.createMap<RealmValue>(mapHandle, null);
+
+          // Necessary since Core will not clear the collection if the value was already a collection
```

Comment:
I wonder if it would be better to track what keys are not updated, and the remove only those, instead of removing all followed by setting them again.

I believe those to operations look quite different in the oplog.

Something similar holds for list, ie. only clear the elements in the old-list from length og the new-list (if any), and update the rest.

Reply:
I guess it also depends on what operations are expected when you e.g. set a field to a new collection. In this case, it may be preferable/expected to have the items cleared, then set new ones, rather than removing some items and updating some items.

Reply:
It is definitely a choice, but it would avoid https://github.com/realm/realm-core/issues/7422.

Reply:
I don't think we want to avoid that though (that issue) as it could make it easier for the SDKs, and the SDKs wouldn't have to call into Core for every item insert. It's definitely an interesting discussion to have, also for Core if we do delegate to them.

Reply:
Eventually the behavior should move to core, as to avoid the repeated calls. I'm more interested in what behavior we want. 

Even if we go with _clear, then assign_, I think this should be handled by core, not the individual SDKs.

Also, to be consistent we should allow setting all collections, not just those in a mixed.

Reply:
Thanks for the clarification! Regarding the last example, for the SDKs having the `Mixed` wrapped (e.g. the `RealmValue` class), I don't think the expectation necessarily is that `myList.add(2);` also should be acting on `myPodo.mixed` due to the intermediate API that they'd have to go through `RealmValue.from()`.

In this case, I'd say that a user's expectations may instead be based on what the API docs of `RealmValue.from()` says that it returns (e.g. the same reference as `myList` or something else (the latter in this case)).

I think the ambiguity of what users would expect becomes more ambiguous for the JS SDK where the syntax is not differentiated as we don't wrap the `Mixed` value.